src/constructs/iam/policies/kcl.ts

```diff
@@ -0,0 +1,65 @@
+import { Effect, PolicyStatement } from "aws-cdk-lib/aws-iam";
+import type { GuStack } from "../../core";
+import { GuPolicy } from "./base-policy";
+
+export interface KCLApplication {
+  streamName: string;
+  applicationName: string;
+}
+
+const kinesisActions: string[] = [
+  "DescribeStream",
+  "DescribeStreamSummary",
+  "RegisterStreamConsumer",
+  "GetRecords",
+  "GetShardIterator",
+  "ListShards",
+];
+
+const kinesisEnhancedFanOutActions: string[] = ["SubscribeToShard", "DescribeStreamConsumer"];
+
+const actionsOnAllTables: string[] = ["Scan", "CreateTable", "DescribeTable", "GetItem", "PutItem"];
```

Comment:
IIRC v2 of the KCL would create the dynamodb table by default, but it was also happy to "find" it, allowing us to create it via IaC. It looks like we know the names of the tables; should we create them ourselves and by extension apply tags to them?

Reply:
> should we create them ourselves and by extension apply tags to them?

Hmm, to create the tables we'd also need to know the DynamoDB primary key & global secondary index structures - by contrast with the IAM permissions and table names, which are explicitly documented, the primary keys & secondary index structure are _not_ - they feel like internal details which AWS are not encouraging us to depend on.

Obviously we could work them out if we wanted to, allowing us to create-&-tag them, but if so, I think it would be reasonable to do that in a subsequent PR!

Reply:
Fair. Wonder if its worth having an inline comment here to explain the `CreateTable` permission?

Reply:
> Wonder if its worth having an inline comment here to explain the `CreateTable` permission?

👍  - done with 2bd57a563e22e9caa118b4171ed9194dae57bbcf!